src/main/java/reposense/model/RepoConfiguration.java

```diff
@@ -124,6 +124,9 @@ public static void setGroupConfigsToRepos(List<RepoConfiguration> repoConfigs,
             List<GroupConfiguration> groupConfigs) {
         for (GroupConfiguration groupConfig : groupConfigs) {
             if (groupConfig.getLocation().isEmpty()) {
+                for (RepoConfiguration repoConfig : repoConfigs) {
```

Comment:
For these lines, it may be better to just set matchingRepoConfig to all the repo, so we reduce code duplicates.